main/precompiled/identity.zkasm

```diff
@@ -1,29 +1,103 @@
 IDENTITY:
-    GAS - 15 => GAS     :JMPN(outOfGas)
+    ; Move balances if value > 0 just before executing the contract CALL
+    $ => B                          :MLOAD(txValue)
+    0 => A
+    zkPC+2 => RR
+    $                               :LT, JMPC(moveBalances)
+    
+    GAS - 15 => GAS :JMPN(outOfGas)
     $ => C          :MLOAD(argsLengthCall)
     ${(C+31)/32} => A
-    GAS - 3*A => GAS    :JMPN(outOfGas)
+    GAS - 3*A => GAS:JMPN(outOfGas)
     CTX             :MSTORE(currentCTX)
-    $ => B          :MLOAD(retCallOffset)
+    CTX => A
     $ => CTX        :MLOAD(originCTX)
-    $ => E          :MLOAD(argsOffsetCall)
+    A               :MSTORE(retDataCTX)
+    A => CTX
+    0 => B
+    0 => E
+    B               :MSTORE(retDataOffset)
+    C               :MSTORE(retDataLength)
 
-IDENTITY32:
-    C - 1           :JMPN(preEnd)
+IDENTITYinit:
+    ; Copy from Stack to memory
+    C - 1           :JMPN(IDENTITYreturn)
     C - 32          :JMPN(IDENTITYfinal)
+    B               :MSTORE(arithA)
+    32              :MSTORE(arithB)
+                    :CALL(divARITH)
+    $ => D          :MLOAD(arithRes2)
+    $ => A          :MLOAD(arithRes1)
+    1024 + A => SP
+    $ => A          :MLOAD(SP)
+                    :CALL(SHLarith)
+    C               :MSTORE(tmpVarC3)
+    A => C
+    $ => A          :MLOAD(arithRes1)
+    1024 + A + 1 => SP
+    32 - D => D
+    $ => A          :MLOAD(SP)
+                    :CALL(SHRarith)
+    A + C           :MSTORE(bytesToStore)
+                    :CALL(MSTORE32)
+    $ => C          :MLOAD(tmpVarC3)
+    C - 32 => C
+    B + 32 => B
+                    :JMP(IDENTITYinit)
+IDENTITYfinal:
+    ${B%32} => D
+    1024 + ${B/32} => SP
+    $ => A          :MLOAD(SP)
+                    :CALL(SHLarith)
+    1024 + ${B/32} + 1 => SP
+    C - 32 + D => D
+    D               :JMPN(IDENTITYxor)
```

Comment:
This code in this case in unreachable bc ${B%32} => D will be always 0
But in the case of the calldata i think the case where C + D is 32 should also go to the same case escenario